### PR TITLE
Filter secondary IP address by type

### LINKED ISSUE #61370: Bug in IPv6 Interfaces. Could add IPv4 secondary Adresses to IPv6 List
There is no filter for the secondary IP Address if it is an IPv4 Address. it would be added to the list of IPv6 Addresses
https://github.com/saltstack/salt/blob/d4b70452df0923648cba480b9d718f555da6f270/salt/grains/core.py#L2555-L2557

### PR DESCRIPTION
**Pull request by mayrstefan**
### What does this PR do?
Only add secondary IP addresses if type belongs to IPv4 oder IPv6 interfaces 

### What issues does this PR fix or reference?
Fixes: #61370

### Previous Behavior
Secondary IPs were added twice: as an IPv4 and IPv6 address because the type field was not evaluated

### New Behavior
Add secondary IP to IPv4 addresses if type is "inet" and to IPv6 addresses if type is "inet6"

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by mayrstefan** [comment]
Sorry for the many commits. It was kind of trial and error to find out what works locally AND the Jenkins test infrastructure. I'm still not sure if I implemented the test in a way that it should be. Too much magic I don't understand ...

File: salt/grains/core.py [hunk 1]
```diff
@@ -2528,7 +2528,7 @@ def ip4_interfaces():
             if "address" in inet:
                 iface_ips.append(inet["address"])
         for secondary in ifaces[face].get("secondary", []):
-            if "address" in secondary:
+            if "address" in secondary and "type" in secondary and secondary["type"] == "inet":
```
Comment (twangboy) [inline salt/grains/core.py:2531]:
```
if "address" in secondary and secondary.get("type") == "inet"
```

File: salt/grains/core.py [hunk 2]
```diff
@@ -2553,7 +2553,7 @@ def ip6_interfaces():
             if "address" in inet:
                 iface_ips.append(inet["address"])
         for secondary in ifaces[face].get("secondary", []):
-            if "address" in secondary:
+            if "address" in secondary and "type" in secondary and secondary["type"] == "inet6":
```
Comment (twangboy) [inline salt/grains/core.py:2556]: same here

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/grains/core.py (hunk 1 above)
Comment (waynew) [reply salt/grains/core.py]: FWIW the `key in thing and thing[key]` approach is *slightly* faster for most cases. Obviously less idiomatic :shrug:

**Review by waynew — APPROVED** [review]
This seems entirely reasonable to me :+1:

**Review by twangboy — APPROVED** [review]

**Comment by mayrstefan** [comment]
I guess this PR somewhat conflicts with the solution in #61061. Which one will be merged?

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**